example/src/Main.purs

```diff
@@ -0,0 +1,181 @@
+module Main where
+
+import Prelude
+
+import Color (Color, rgb)
+import ColorPicker.Halogen.Component as CPicker
+import ColorPicker.Halogen.Layout as L
+import Control.Monad.Aff.Class (class MonadAff)
+import Control.Monad.Eff (Eff)
+import Control.MonadZero (guard)
+import Data.Array (reverse)
+import Data.Either.Nested as Either
+import Data.Functor.Coproduct.Nested as Coproduct
+import Data.Map (Map, insert, lookup)
+import Data.Maybe (Maybe(..), maybe')
+import Data.Monoid (mempty)
+import Halogen as H
+import Halogen.Aff as HA
+import Halogen.Component.ChildPath as CP
+import Halogen.HTML as HH
+import Halogen.HTML.Events as HE
+import Halogen.HTML.Properties as HP
+import Halogen.VDom.Driver (runUI)
+
+main ∷ Eff (HA.HalogenEffects ()) Unit
+main = HA.runHalogenAff do
+  body ← HA.awaitBody
+  runUI example unit body
+
+data Query a = HandleMsg ColorIdx CPicker.Message a
+
+type State = Map Int {current ∷ Color, next ∷ Color }
+type ColorIdx = Int
+type ChildQuery = Coproduct.Coproduct1 CPicker.Query
+type Slot = Either.Either1 ColorIdx
+
+
+cpColor ∷ CP.ChildPath CPicker.Query ChildQuery ColorIdx Slot
+cpColor = CP.cp1
+
+
+type HTML m = H.ParentHTML Query ChildQuery Slot m
+type DSL m = H.ParentDSL State Query ChildQuery Slot Void m
+
+
+example ∷ ∀ m r. MonadAff (CPicker.PickerEffects r) m => H.Component HH.HTML Query Unit Void m
+example = H.parentComponent
+    { initialState: const mempty
+    , render
+    , eval
+    , receiver: const Nothing
+    }
+
+render ∷ ∀ m r. MonadAff (CPicker.PickerEffects r) m => State → HTML m
+render state = HH.div_
+  $ renderPicker 0 config0
+  <> renderPicker 1 config1
+  <> renderPicker 2 config2
+
+  where
+  renderPicker idx conf =
+    [ HH.h1_ [ HH.text $ "Picker " <> show idx ]
+    , HH.slot' cpColor idx CPicker.picker conf (HE.input $ HandleMsg idx)
+    , HH.p_ [ HH.text case lookup idx state of
+        Just ({current, next}) →
+          "uncommited (current: " <> show current <>", next:" <> show next <> ")"
+        Nothing → "no color"
+      ]
+    ]
+
+eval ∷ ∀ m. Query ~> DSL m
+eval (HandleMsg idx msg next) = do
+  H.modify update
+  pure next
+  where
+  update state = insert idx val state
+    where
+    val = case lookup idx state, msg of
+      Just s, CPicker.NextChange nextVal → s{next = nextVal}
+      _, CPicker.NextChange x → { next: x, current: x }
+      _, CPicker.NotifyChange x →  {next: x, current: x}
+
+config0 ∷ CPicker.Props
+config0 = mkConf $ L.Root c $ reverse l
+  where
+  L.Root c l = mkLayout
+    (H.ClassName "ColorPicker--small")
```

Comment:
it's BEM style naming. more specificaly [suitcss](https://github.com/suitcss/suit/blob/master/doc/naming-conventions.md).

This doc with it's interactive walkthrough is pretty good:
https://github.com/gilbox/css-bliss
http://gilbox.github.io/css-bliss/walkthrough/release/